tests/test-cases/Group23-VIC-Machine-Service/23-06-VCH-Certificate.robot

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+*** Settings ***
+Documentation  Test 23-06 - VCH Certificate
+Resource  ../../resources/Util.robot
+Suite Setup  VIC Machine Server Test Setup
+Suite Teardown  VIC Machine Server Test Teardown
+Default Tags
+
+*** Keywords ***
+VIC Machine Server Test Setup
+    Install VIC Appliance To Test Server
+    Start VIC Machine Server
+
+VIC Machine Server Test Teardown
+    Cleanup VIC Appliance On Test Server
+    Terminate All Processes  kill=True
+
+Start VIC Machine Server
+    Start Process  ./bin/vic-machine-server --port 31337  shell=True  cwd=/go/src/github.com/vmware/vic
+
+Curl No Datacenter
+    [Arguments]  ${vch-id}  ${auth}
+    ${rc}  ${output}=  Run And Return Rc And Output  curl -s -X GET "http://127.0.0.1:31337/container/target/%{TEST_URL}/vch/${vch-id}/certificate?thumbprint=%{TEST_THUMBPRINT}" -H "authorization: Basic ${auth}"
+    [Return]  ${rc}  ${output}
+
+Curl Datacenter
+    [Arguments]  ${vch-id}  ${auth}
+    ${orig}=  Get Environment Variable  TEST_DATACENTER
+    ${dc}=  Run Keyword If  '%{TEST_DATACENTER}' == '${SPACE}'  Get Datacenter Name
+    Run Keyword If  '%{TEST_DATACENTER}' == '${SPACE}'  Set Environment Variable  TEST_DATACENTER  ${dc}
+    ${rc}  ${output}=  Run And Return Rc And Output  curl -s -X GET "http://127.0.0.1:31337/container/target/%{TEST_URL}/datacenter/%{TEST_DATACENTER}/vch/${vch-id}/certificate?thumbprint=%{TEST_THUMBPRINT}" -H "authorization: Basic ${auth}"
+    Set Environment Variable  TEST_DATACENTER  ${orig}
+    [Return]  ${rc}  ${output}
+
+*** Test Cases ***
+Get VCH Certificate
+    ${id}=  Get VCH ID  %{VCH-NAME}
+    ${auth}=  Evaluate  base64.b64encode("%{TEST_USERNAME}:%{TEST_PASSWORD}")  modules=base64
+    ${rc}  ${output}=  Curl No Datacenter  ${id}  ${auth}
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  BEGIN CERTIFICATE
+    Should Contain  ${output}  END CERTIFICATE
+    ${rc}  ${outputDC}=  Curl Datacenter  ${id}  ${auth}
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  BEGIN CERTIFICATE
+    Should Contain  ${output}  END CERTIFICATE
```

Comment:
Maybe also assert that the certificate doesn't contain any double quotation marks? (That seems like a decent approximation of asserting that we got text, not a JSON object.)